doc/readme.md

```diff
@@ -0,0 +1,335 @@
+- `frame` wird in jedem Frame einmal aufgerufen.
+- `main` startet das Spiel.
+
+- `entities`
+-entitiestoadd
+-entitiestoremove
+
+### Component
+
+- Components speichern immer die Entität zu der sie gehören
+- alle bereits implementierten Components finden Sie [hier](/ecs/components/readme.md)
+- um eigene Components zu schreiben, leiten Sie von der eigentlichen Component Klasse ab
+- siehe [create_own_content.md](ecs/create_own_content.md)
+
+### Entity
+
+- Entitäten sind leere Container und speichern Components
+- um ein Component einer Entität zu erhalten, benutzen Sie die Methode `entity#getComponent(component.class klass)`
+    - Anmerkung: Sie erhalten ein `Optional zurück`
+- siehe [create_own_content.md](doc/ecs/create_own_content.md)
+
+### System
+
+- um eigene Components zu schreiben, leiten Sie von der ECS_Systems Klasse ab
+- in der `ECS_System#update()` iterieren die Systeme über alle Entitäten mit bestimmten Components und agieren darauf
+- alle bereits implementierten Systeme finden Sie [hier](ecs/systems/readme.md)
+- siehe [create_own_content.md](ecs/create_own_content.md)
+
+
+## Übung: Eigenen Helden erstellen
+
+Zwar gibt es in den Vorlagen bereits einen Helden (den schauen wir uns am Ende dieses Kapitels genauer an), trotzdem wird Ihnen hier erklärt, wie Sie Ihre erste eigene Entität in das Spiel implementieren.
+
+*Hinweis: Wenn Sie mitprogrammieren wollen, löschen Sie die Klasse `ecs.entities.Hero` und löschen Sie in `starter.Game` die Zeile 118 (`hero = new Hero(new Point(0, 0));`) und das Import-Statement `import ecs.entities.Hero;`.*
+
+### Held Entität erstellen
+
+Zu Beginn erstellen Sie sich eine neue Java-Klasse namens `MyHero`. Diese erbt von der Klasse `Entity`.
+
+``` java
+package ecs.entities;
+
+public class MyHero extends Entity{}
+```
+
+Der Held ist ein Element im Spiel und daher eine Entität im ECS.
+
+Damit der Held im Level platziert werden kann, braucht er ein `PositionComponent`. Das `PositionComponent` speichert den `Point` (also die Position) auf dem sich der Held befindet.
+
+Im Dungeon existieren zwei Koordinatensysteme. Die Level werden als Matrix von `Tile`s gespeichert. `Tile`s sind die Felder im Level (Boden, Wand, Loch, etc.). Mehr zum Level finden Sie [hier](level/readme.md). Die Position der `Tile`s werden als `Coordinate`s gespeichert (Index des Tiles in der Matrix). Entitäten können auch zwischen zwei Tiles stehen. Daher werden ihre Positionen als `Point`s gespeichert.
+
+Im Konstruktur des Helden legen wir das `PositionComponent` an. Dazu erzeugen wir ein neues `PositionComponent` mit der Startposition `(0|0)`
+
+``` java
+package ecs.entities;
+
+import ecs.components.PositionComponent;
+import tools.Point;
+
+public class MyHero extends Entity {
+
+    public MyHero() {
+        super();
+        new PositionComponent(this, new Point(0, 0));
```

Comment:
@AMatutat Kriegen wir hier einen Convenience-Konstruktor hin, so dass man schreiben kann `new PositionComponent(this);`? Oder wenigstens `new PositionComponent(this,0,0);`?

Reply:
> Kriegen wir hier einen Convenience-Konstruktor hin, so dass man schreiben kann new PositionComponent(this);?

ich glaube ich kann da was basteln

Reply:
#545